src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -357,4 +398,61 @@ public void disconnect() {
         void setProxy(URL proxyUrl);
     }
 
+    /**
+     * An InputStreamGenerator has a single method getInputStream. Implementors return an
+     * InputStream ready for consuming by the HttpConnection. The purpose of this is to
+     * facilitate regeneration of an InputStream for cases where a payload is going to be resent
+     * for a retry. Implementors must not return the same InputStream instance from multiple calls
+     * to {@link #getInputStream()} unless the stream has been reset.
+     */
+    public interface InputStreamGenerator {
+        /**
+         * @return an InputStream to use to read the body content for a HTTP request
```

Comment:
no `@throws` tag present, should probably have a summary too even though its only a getter